### PR TITLE
Exclude test results by test_id or testrun_id

### PR DESCRIPTION
**Pull request by aleksbykov**
before run check regression, find provided test results by test run or
test id, and mark them as excluded and not use in compare for Best and Last

### CODE REVIEW

File: sdcm/microbenchmarking.py [hunk 1]
```diff
@@ -24,11 +24,15 @@
 
 
 class MicroBenchmarkingResultsAnalyzer(BaseResultsAnalyzer):
-    def __init__(self, email_recipients):
+    def __init__(self, email_recipients=None, cur_version=None):
+        send_email = False
```
[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/microbenchmarking.py]: fixed

File: sdcm/microbenchmarking.py [hunk 2]
```diff
@@ -51,13 +56,16 @@ def check_regression(self, current_results, html_report_path):
...
 
         cur_version_info = current_results[current_results.keys()[0]]['versions']['scylla-server']
         cur_version = cur_version_info["version"]
         tests_filtered = self._es.search(index=self._es_index, filter_path=filter_path, size=self._limit,
-                                         q="hostname:'%s' AND versions.scylla-server.version:%s*" % (self.hostname,
-                                                                                                     cur_version[:3]))
+                                         q="hostname:'%s' \
+                                            AND versions.scylla-server.version:%s* \
+                                            AND ((-_exists_:excluded) OR (excluded:false))" % (self.hostname,
+                                                                                               cur_version[:3]))
```
[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/microbenchmarking.py]: Fix

File: sdcm/microbenchmarking.py [hunk 3]
```diff
@@ -231,34 +240,169 @@ def get_results(self, results_path, update_db):
...
+        self.log.info('Exclude tests before date {}'.format(date))
+        results = self._es.search(index=self._es_index, filter_path=filter_path, size=self._limit,
+                                  q="hostname:'%s' AND versions.scylla-server.version:%s*" %
+                                  (self.hostname, self.cur_version[:3]))
+        if not results:
+            self.log.info('Nothing to exclude')
+            return
+
+        before_date_results = [
+            doc for doc in results['hits']['hits']
```
[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/microbenchmarking.py]: fix

File: sdcm/microbenchmarking.py [hunk 4]
```diff
@@ -37,6 +41,7 @@ def __init__(self, email_recipients):
         self.hostname = socket.gethostname()
         self._run_date_pattern = "%Y-%m-%d_%H:%M:%S"
         self.test_run_date = datetime.datetime.now().strftime(self._run_date_pattern)
+        self.cur_version = cur_version
```
[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/microbenchmarking.py]: fix

File: sdcm/microbenchmarking.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bentsi) [reply sdcm/microbenchmarking.py]: still see the code here :) or maybe I misunderstood smth

[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/microbenchmarking.py]: Sorry, think about another part of code

**Pull request merged**